subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationQueue.java

```diff
@@ -138,77 +127,85 @@ private static String getFailureMessage(Collection<? extends Throwable> failures
         return "Multiple build operations failed.";
     }
 
-    private class OperationHolder implements Runnable {
-        private final WorkerLeaseRegistry.WorkerLease parentWorkerLease;
-        private final T operation;
-        private BuildOperationDescriptor operationDescription;
+    private class WorkerRunnable implements Runnable {
+        @Override
+        public void run() {
+            while (true) {
```

Comment:
Style issue: instead of `while(true)`, you could have:

```
while ((operation = waitForNextOperation())!=null) {
   runBatch(operation);
}
shutDown();
```